add.go

```diff
@@ -74,9 +76,27 @@ type AddAndCopyOptions struct {
 	StripStickyBit bool
 }
 
-// sourceIsRemote returns true if "source" is a remote location.
+// gitURLFragmentSuffix matches fragments to use as Git reference and build
+// context from the Git repository e.g.
+//
+//	github.com/containers/buildah.git
+//	github.com/containers/buildah.git#main
+//	github.com/containers/buildah.git#v1.35.0
+var gitURLFragmentSuffix = regexp.MustCompile(`\.git(?:#.+)?$`)
```

Comment:
Not sure this is necessary, and MustCompile on init variables slows things down on startup.

Reply:
can we use  `Delayed()` from the `containers/storage/pkg/regexp` package?

Reply:
Using `Delayed()` now, it seems to help with your concern @rhatdan, wdyt?